### PR TITLE
feat(Core/Logging): improved conditions error log messages

### PR DESCRIPTION
**Pull request by IntelligentQuantum**
## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes 

## SOURCE:

https://github.com/TrinityCore/TrinityCore/commit/d9c6b8fd89514e2b31dc1c4ac6668b5374d51fc1
## Tests Performed:

- 
- 

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Nyeriah** [comment]
Hey, question: this doesn't handle all of our source types, what happens if the source type string array doesn't have the definition for the AC source type?

File: src/server/game/Conditions/ConditionMgr.cpp [hunk 1]
```diff
@@ -1275,32 +1374,40 @@ bool ConditionMgr::isSourceTypeValid(Condition* cond)
...
-        LOG_ERROR("sql.sql", "Invalid ConditionSourceType %u in `condition` table, ignoring.", uint32(cond->SourceType));
+        LOG_ERROR("sql.sql", "%s Invalid ConditionSourceType in `condition` table, ignoring.", cond->ToString().c_str());
         return false;
     }
 
     switch (cond->SourceType)
     {
     case CONDITION_SOURCE_TYPE_TERRAIN_SWAP:
+    {
+        LOG_ERROR("sql.sql", "CONDITION_SOURCE_TYPE_TERRAIN_SWAP: is only for master branch, skipped");
```
Comment (Winfidonarleyan) [inline src/server/game/Conditions/ConditionMgr.cpp:1385]: ?

Comment (IntelligentQuantum) [reply src/server/game/Conditions/ConditionMgr.cpp:1385]: PR not completed yet i add all of this shit then i start cleanup xD

File: src/server/game/Conditions/ConditionMgr.h [hunk 2]
```diff
@@ -259,14 +262,24 @@ class ConditionMgr
...
+    [[nodiscard]] static bool CanHaveSourceIdSet(ConditionSourceType sourceType);
     ConditionList GetConditionsForNotGroupedEntry(ConditionSourceType sourceType, uint32 entry);
     ConditionList GetConditionsForSpellClickEvent(uint32 creatureId, uint32 spellId);
     ConditionList GetConditionsForSmartEvent(int32 entryOrGuid, uint32 eventId, uint32 sourceType);
     ConditionList GetConditionsForVehicleSpell(uint32 creatureId, uint32 spellId);
     ConditionList GetConditionsForNpcVendorEvent(uint32 creatureId, uint32 itemId);
 
+    struct ConditionTypeInfo
+    {
+        char const* Name;
```
Comment (Winfidonarleyan) [inline src/server/game/Conditions/ConditionMgr.h:275]: can use `std::string` ?

Comment (IntelligentQuantum) [reply src/server/game/Conditions/ConditionMgr.h:275]: Yes but after port all conditions

**Comment by Kitzunu** [comment]
status?

**Comment by IntelligentQuantum** [comment]
WIP

**Comment by acidmanifesto** [comment]
What is the current status of this? Is it abandoned?

**Comment by acidmanifesto** [comment]
Final Request for Status of this PR?

**Comment by IntelligentQuantum** [comment]
@acidmanifesto I think the label shows the status of the pull request, And if you want to work on this, I will close it

**Comment by IntelligentQuantum** [comment]
I will continue when I have free time, don't worry

**Pull request closed without merging**